File: backend/src/main/java/com/example/backend/model/Transaction.java
package com.example.backend.model;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@AllArgsConstructor
@EqualsAndHashCode
@ToString
public class Transaction {

    private final LocalDate transactionDate;
    private final String accountNumber;
    private final String accountHolderName;
    private final BigDecimal amount;
    private final TransactionStatus status;
}

File: frontend/types/transaction.ts
export type TransactionStatus = "Pending" | "Settled" | "Failed"
export type UIState = "ready" | "loading" | "empty" | "error"

export interface Transaction {
  transactionDate: string
  accountNumber: string
  accountHolderName: string
  amount: number
  status: TransactionStatus
}
